build_test.go

```diff
@@ -0,0 +1,285 @@
+package composer_test
+
+import (
+	"bytes"
+	. "github.com/onsi/gomega"
+	"github.com/paketo-buildpacks/composer"
+	"github.com/paketo-buildpacks/composer/fakes"
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/draft"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/postal"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+	"github.com/sclevine/spec"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func testBuild(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+
+		cnbDir     string
+		workingDir string
+		layersDir  string
+		dependency postal.Dependency
+
+		buffer            *bytes.Buffer
+		dependencyManager *fakes.DependencyManager
+		entryResolver     = draft.NewPlanner()
+
+		build     packit.BuildFunc
+		buildPlan packit.BuildpackPlan
+	)
+
+	it.Before(func() {
+		buffer = bytes.NewBuffer(nil)
+		logEmitter := scribe.NewEmitter(buffer)
+
+		var err error
+		layersDir, err = os.MkdirTemp("", "layers")
+		Expect(err).NotTo(HaveOccurred())
+
+		workingDir, err = os.MkdirTemp("", "working-dir")
+		Expect(err).NotTo(HaveOccurred())
+
+		cnbDir, err = os.MkdirTemp("", "cnb")
+		Expect(err).NotTo(HaveOccurred())
+
+		dependencyManager = &fakes.DependencyManager{}
+
+		build = composer.Build(logEmitter, dependencyManager, entryResolver)
+
+		composerArchive, err := os.CreateTemp(cnbDir, "composer-archive")
+		Expect(err).NotTo(HaveOccurred())
+		composerArchiveName := filepath.Base(composerArchive.Name())
+
+		Expect(os.Chmod(composerArchive.Name(), 0777)).To(Succeed())
+
+		dependency = postal.Dependency{
+			ID:      "composer",
+			Name:    composerArchiveName,
+			Version: "composer-dependency-version",
+			SHA256:  "some-sha",
+		}
+
+		dependencyManager.ResolveCall.Returns.Dependency = dependency
+		dependencyManager.DeliverCall.Stub = func(dependency postal.Dependency, cnbPath, layerPath, _ string) error {
+			return fs.Copy(filepath.Join(cnbPath, dependency.Name), filepath.Join(layerPath, dependency.Name))
+		}
+
+		buildPlan = packit.BuildpackPlan{
+			Entries: []packit.BuildpackPlanEntry{
+				{
+					Name: "composer",
+					Metadata: map[string]interface{}{
+						"build":  true,
+						"launch": true,
+					},
+				},
+			},
+		}
+	})
```

Comment:
Two things:
1. Why not use the `entryResolver.ResolveCall.Returns...` feature here instead to take advantage of the fakes?
2. Be careful with conflating `build plan` ([spec definition](https://github.com/buildpacks/spec/blob/main/buildpack.md#build-plan-toml)) and `buildpack plan` ([CNB spec definition](https://github.com/buildpacks/spec/blob/main/buildpack.md#buildpack-plan-toml)). They're two different things.